integration/upload_test.go

```diff
@@ -0,0 +1,250 @@
+	}
+	upload := newHTTPUpload(suite.T(), suite.cfg.HTTPServer)
+	suite.testUpload(upload)
+}
+
+func (suite *uploadTestSuite) TestAzureUpload() {
+	options, err := uploaders.GetAzureTestOptions(suite.T())
+	require.NoError(suite.T(), err, "error getting azure test options")
+	upload := newAzureUpload(suite.T(), options)
+	suite.testUpload(upload)
+}
+
+func (suite *uploadTestSuite) TestAWSUpload() {
+	options := uploaders.GetAWSTestOptions(suite.T())
+	upload, err := newAWSUpload(suite.T(), options)
+	require.NoError(suite.T(), err, "error creating AWS client")
+	suite.testUpload(upload)
+}
+
+func (suite *uploadTestSuite) triggerUploads() map[string]string {
+	requestPath := fmt.Sprintf("/features/%s/outbox/messages/request", featureID)
+	requestedFiles := make(map[string]string)
+
+	connMessages, err := util.NewDigitalTwinWSConnection(suite.initializer.Cfg)
+	require.NoError(suite.T(), err, "Failed to create websocket connection")
+	defer connMessages.Close()
+
+	suite.startListening(connMessages, typeMessages)
+	url := fmt.Sprintf("%s/inbox/messages/%s", suite.featureURL, operationTrigger)
+	_, err = util.SendDigitalTwinRequest(suite.initializer.Cfg, http.MethodPost, url, map[string]interface{}{
+		paramCorrelationID: "test",
+	})
+	require.NoError(suite.T(), err, "error sending digital twin request for trigger operation")
+	err = util.ProcessWSMessages(suite.initializer.Cfg, connMessages,
+		func(event *protocol.Envelope) (bool, error) {
+			if requestPath == event.Path {
+				request := &uploadRequest{}
+				err := parseEventValue(event.Value, request)
+				if err != nil {
+					suite.T().Logf("cannot convert %v to upload request, will ignore the event", event.Value)
+					return false, nil
+				}
+				if request.Options == nil {
+					suite.T().Logf("no upload request options found in payload(%v)", request)
+					return false, nil
+				}
+				if path, ok := request.Options["file.path"]; ok {
+					suite.T().Logf("file upload request: %s, with correlation id: %s", path, request.CorrelationID)
+					requestedFiles[request.CorrelationID] = path
+					return len(requestedFiles) == uploadFilesCount, nil
+				}
+				suite.T().Log("file.path key not found in upload request event options")
+			}
+			return len(requestedFiles) == uploadFilesCount, nil
+		})
+
+	require.NoError(suite.T(), err, "error processing file upload requests")
+	require.Equal(suite.T(), uploadFilesCount, len(requestedFiles), "wrong file upload request events count")
+	suite.T().Logf("%v file upload requests, initiating uploads", len(requestedFiles))
+	return requestedFiles
+}
+
+func (suite *uploadTestSuite) startUploads(testUpload upload, requestedFiles map[string]string, files []string) {
+	connEvents, err := util.NewDigitalTwinWSConnection(suite.initializer.Cfg)
+	require.NoError(suite.T(), err, "Failed to create websocket connection")
+	defer connEvents.Close()
+
+	suite.startListening(connEvents, typeEvents)
+
+	requestedFilesRev := make(map[string]string)
```

Comment:
`Rev` - what does it mean here?

Reply:
Will be renamed to something more convenient.